Translate "Hooks" index page (#618)
reactjs/ja.react.dev: src/content/reference/react/index.md

  // ...
```

エフェクトは、React パラダイムからの「脱出ハッチ」のようなものです。エフェクトをアプリケーションのデータフローを調整するために使ってはいけません。外部のシステムとやりとりを行わないならば、[エフェクトは必要ないかもしれません。](/learn/you-might-not-need-an-effect)

> `useEfefct` には、タイミングの違いによってまれに使われることのある 2 つのバリエーションがあります： 

Reviewer: ```suggestion
`useEffect` には、実行タイミングが異なり、まれに使われることのある 2 つのバリエーションがあります： 
```

* [`useLayoutEffect`](/reference/react/useLayoutEffect) はブラウザが画面を再描画する前に発火します。このフックでレイアウトを測定できます。
* [`useInsertionEffect`](/reference/react/useInsertionEffect) は React が DOM に変更を加える前に発火します。ライブラリはダイナミック CSS をこのフックで挿入できます。

---

## performance フック {/*performance-hooks*/}

> 再レンダーのパフォーマンスを最適化する通常の方法は、不要な処理を減らすことです。例えばキャッシュを再利用したり、データの変更がない場合の再レンダーをスキップしたりするよう、React に伝えることができます。

Reviewer: ```suggestion
再レンダーのパフォーマンスを最適化するためのよくある方法は、不要な処理を減らすことです。例えばキャッシュ済みの計算結果を再利用したり、データの変更がない場合の再レンダーをスキップしたりするよう、React に伝えることができます。
```

キャッシュとは仕組みを指す言葉なので、キャッシュ自体を再利用することはできません

}
```

画面を更新するために再レンダーをスキップできない場合もあるでしょう。その場合、同期が必要なブロック更新（ユーザの文字入力など）を、ユーザ インターフェイスをブロックする必要のない非ブロック更新（図の更新など）から分離することで、パフォーマンスを向上することができます。

レンダリングを優先するには、これらのフックを用いることができます： 

- [`useTransition`](/reference/react/useTransition) を用いると、状態の遷移を非ブロックとしてマークし、他のアップデートによる割り込みを許可します。
- [`useDeferredValue`](/reference/react/useDeferredValue) を用いると、UI の重要でない部分の更新を延期して他の部分を先に更新させることができます。

---

## その他のフック {/*other-hooks*/}

> これらのフックはライブラリの開発者には便利かもしれません。しかし、アプリケーションのコードでは通常は用いられることはありません。

Reviewer: ```suggestion
これらのフックはライブラリの開発者には有用ですが、アプリケーションコードでは通常は用いられることはありません。
```

直訳で問題ないので1文にまとめました

- [`useDebugValue`](/reference/react/useDebugValue) を用いると、React DevTools が表示するカスタムフックのラベルをカスタマイズできます。

> - [`useId`](/reference/react/useId) を用いると、コンポーネントがユニークな ID をコンポーネントそのものに関連付けることができます。通常はアクセシビリティ API とともに使用されます。 

Reviewer: ```suggestion
- [`useId`](/reference/react/useId) を用いると、コンポーネントにユニークな ID を関連付けることができます。通常はアクセシビリティ API とともに使用されます。 
```

- [`useSyncExternalStore`](/reference/react/useSyncExternalStore) を用いると、コンポーネントは外部のストアを参照できるようになります。

---

## 独自のフック {/*your-own-hooks*/}

> 独自のカスタムフックを[ JavaScript の関数として定義](/learn/reusing-logic-with-custom-hooks#extracting-your-own-custom-hook-from-a-component)することもできます。

Reviewer: ```suggestion
JavaScript の関数として[独自のカスタムフックを定義](/learn/reusing-logic-with-custom-hooks#extracting-your-own-custom-hook-from-a-component)することもできます。
```
